src/dialogs/more-info/controls/more-info-siren.ts

```diff
@@ -0,0 +1,130 @@
+import "@material/mwc-button/mwc-button";
+import { css, CSSResultGroup, html, LitElement, TemplateResult } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { stopPropagation } from "../../../common/dom/stop_propagation";
+import { supportsFeature } from "../../../common/entity/supports-feature";
+import "../../../components/ha-duration-input";
+import { UNAVAILABLE } from "../../../data/entity";
+import { SirenEntity, SirenEntityFeature } from "../../../data/siren";
+import { HomeAssistant } from "../../../types";
+
+const DEFAULT_TONE_KEY = "_";
+
+@customElement("more-info-siren")
+class MoreInfoSiren extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public stateObj?: SirenEntity;
+
+  @state() _selectedTone?: string;
+
+  private _toneChanged(ev) {
+    const newVal = ev.target.value;
+
+    if (newVal === DEFAULT_TONE_KEY) {
+      delete this._selectedTone;
+      return;
+    }
+
+    this._selectedTone = newVal;
+
+    if (this.stateObj?.state === "on") {
+      this._turnOn();
+    }
+  }
+
+  private _turnOn() {
+    const data: Record<string, any> = {
+      entity_id: this.stateObj!.entity_id,
+    };
+    if (this._selectedTone) {
+      data.tone = this._selectedTone;
+    }
+    this.hass.callService("siren", "turn_on", data);
+  }
+
+  private _turnOff() {
+    this.hass.callService("siren", "turn_off", {
+      entity_id: this.stateObj!.entity_id,
+    });
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this.stateObj) {
+      return html``;
+    }
+
+    const stateObj = this.stateObj;
+
+    return html`
+      <hr />
+      <div class="actions">
+        ${supportsFeature(stateObj, SirenEntityFeature.TONES) &&
+        this.stateObj.attributes.available_tones
```

Comment:
If we can only change it when it is on, we should only show it when on right? We do the same for a light and effects.

Reply:
A use case can be : I want to turn on the siren with this ringtone.

Reply:
but it now only does something when it is on already?